modules/sor/sorV2/lib/static.ts

```diff
@@ -28,11 +33,23 @@ export async function sorGetSwapsWithPools(
     for (const prismaPool of prismaPools) {
         switch (prismaPool.type) {
             case 'WEIGHTED':
-                basePools.push(WeightedPool.fromPrismaPool(prismaPool));
+                {
+                    if (prismaPool.protocolVersion === 2) {
+                        basePools.push(WeightedPool.fromPrismaPool(prismaPool));
+                    } else {
+                        basePools.push(WeightedPoolV3.fromPrismaPool(prismaPool));
+                    }
+                }
                 break;
             case 'COMPOSABLE_STABLE':
             case 'PHANTOM_STABLE':
-                basePools.push(ComposableStablePool.fromPrismaPool(prismaPool));
+                {
+                    if (prismaPool.protocolVersion === 2) {
+                        basePools.push(ComposableStablePool.fromPrismaPool(prismaPool));
+                    } else {
+                        basePools.push(StablePool.fromPrismaPool(prismaPool));
```

Comment:
@franzns - v3 stable pools actually have their type as 'STABLE', right? I'll update this.

Reply:
Correct.